archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterGenericMethodSignaturesTest.java

```diff
@@ -0,0 +1,1016 @@
+package com.tngtech.archunit.core.importer;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.Serializable;
+import java.lang.ref.Reference;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+import com.google.common.collect.FluentIterable;
+import com.tngtech.archunit.ArchConfiguration;
+import com.tngtech.archunit.base.Function;
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.domain.JavaCodeUnit;
+import com.tngtech.archunit.testutil.ArchConfigurationRule;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static com.tngtech.archunit.testutil.Assertions.assertThat;
+import static com.tngtech.archunit.testutil.Assertions.assertThatCodeUnit;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteClass.concreteClass;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteGenericArray.genericArray;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteGenericArray.parameterizedTypeArrayName;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteGenericArray.typeVariableArrayName;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteParameterizedType.parameterizedType;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteTypeVariable.typeVariable;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteWildcardType.wildcardType;
+import static com.tngtech.java.junit.dataprovider.DataProviders.crossProduct;
+import static com.tngtech.java.junit.dataprovider.DataProviders.testForEach;
+
+@RunWith(DataProviderRunner.class)
+public class ClassFileImporterGenericMethodSignaturesTest {
+
+    @DataProvider
+    public static Object[][] data_of_imports_empty_list_of_type_parameters_for_non_generic_code_unit() {
+        @SuppressWarnings("unused")
+        class TypeParameterOnConstructor {
```

Comment:
I still like "TypeParameterOn", since it makes it clear what's relevant, but I agree that it is weird in the first case :wink: I adjusted it to `NoTypeParameterOn...`!